rfd/0065-distributed-tracing.md

```diff
@@ -0,0 +1,247 @@
+---
+authors: Tim Ross (tim.ross@goteleport.com)
+state: draft
+---
+
+# RFD 65 - Distributed Tracing
+
```

Comment:
```suggestion
# Required Approvers
Engineering: @zmb3 (@fspmarshall || @espadolini)
Product: (@klizhentas || @xinding33)
```

@rosstimothy @r0mant @zmb3 Do we have any other distributed tracing experts on the team that might be good required reviewers? Maybe @codingllama?

Reply:
I'm no tracing expert, I've used a couple tools over time but it's been a while. Happy to take a closer look if you like.

FWIW, I think Tim's overall proposal is a no-brainer, with OpenTelemetry and propagating spans via context.Context. Automatic tracers and boundaries are already much better than nothing; in other scenarios I think manual instrumentation is the way to get good spans. I wouldn't aim at instrumenting everything, but we can do a few codepaths we care about address others as the need arises.

Reply:
I'd be happy to come onto this as an approver/reviewer, I've done a fair bit with Observability in previous roles.